docker/Dockerfile

```diff
@@ -55,8 +41,27 @@ RUN INFER_VERSION=v0.10.0; \
 
 # Compile Infer
 RUN cd /infer && \
-    ./build-infer.sh
+    ./build-infer.sh && \
```

Comment:
I think this can be a bit more efficient because here we're compiling OCaml twice. Instead, you can set up the switch in advance and then tell `./build-infer.sh` that that switch has already been configured and is to be used to build infer:
```sh
RUN OCAML_VERSION=4.04.0; \
    opam init --compiler=$OCAML_VERSION -j $(getconf _NPROCESSORS_ONLN || echo 1) --yes && \
    cd /infer && \
    ./build-infer.sh --opam-switch $OCAML_VERSION && \
    rm -rf /root/.opam
```

(We don't do this by default to ensure the infer install is somewhat sandboxed, although I imagine we could do something like that when we detect that opam is not set up at all...)